### DISCUSSION TITLE
Cell Edit - Input Validation

### DISCUSSION THREAD

**Discussion by mrcwcurtis** [shlomiassaf/ngrid#191] [Q&A]
I understand that the grid supports cell edit functionality. Are there any recommended ways to implement validations for the inputs?

Thanks,
Casey

**Comment by shlomiassaf** [comment]
There are so many ways to do that...

I assume you refer to forms, cause this is how you can validate in angular unless you want to wrap input controls from scratch..

Basically, it depends on a lot of factors. If there's virtual scroll running, infinite scroll, pagination? 
In that case we need context, but if not we still want context to make it work similar in all scenarios
You might create a plugin directive that hooks into ngrid and creates a global form and then for each line edited creates a form for the row and the proper controls, this way you can support virtual/infinite scenarios
In any case, I wouldn't provide the form in advance, just the row's (data) and create the logic internally.
Remember that each cell might have different behaviour and each grid has different columns, so this is where you might want to introduce "type" on the column.
If you want to work with external libs (e.g. formly) you can add a new property to the column (TS augmentation) and pass it to formly (again, top level directive plugin for ngrid) which will manage everything...
Think adapter for ngrid and formly
So so so many ways to solve this...
Btw, ngrid provides the basic tools to edit a cell, integration with forms is something not related to ngrid, at least not the core... Again since there are so many permutations here, it's hard to build a single solution.
And i'm not even addressing UI here, which we need to hook up in someway, externally
